pkg/server/event_handler_test.go

```diff
@@ -325,18 +348,40 @@ func newTestServer(handler func(w http.ResponseWriter, r *http.Request)) (*http.
 	}
 }
 
+// Returns fake metadata that include github resource info.
+func getFakeMetadata() []byte {
+	return []byte(`{
+		"metadata": {
+		  "git-commit": "test-git-commit",
+		  "triggered-timestamp": "test-timestamp",
+		  "git-workflow-sha": "test-workflow-sha",
+		  "git-workflow": "test-workflow",
+		  "git-repo": "test-git-repo"
+		}
+	  }`)
+}
+
 // Returns a fake http func that writes the data in http response.
-func handleObjectRead(t *testing.T, data []byte) func(w http.ResponseWriter, r *http.Request) {
+func handleObjectRead(t *testing.T, data, metadata []byte) func(w http.ResponseWriter, r *http.Request) {
```

Comment:
I personally think it would be better if we can write a fake server like https://github.com/abcxyz/pmap/blob/41c46ce082d64d08396dd60bb3b5a48762f08bdd/pkg/mapping/processors/assetinventory_test.go#L79 for UT. The way to write unit test is a little bit difficulty to understand and implement as we need to figure out the url of gcs function call. And the switch statement of different url is a little bit cumbersome. But no a block for this PR. We can create another issue to improve this.

Reply:
Created issue #95 to solve this.